src/core/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import cx from 'classnames';
+import { useTheme } from '../utils/hooks/useTheme';
+import { CommonProps } from '../utils/props';
+import SvgChevronRight from '@itwin/itwinui-icons-react/cjs/icons/ChevronRight';
+import '@itwin/itwinui-css/css/breadcrumbs.css';
+import { useResizeObserver } from '../utils/hooks/useResizeObserver';
+import { useMergedRefs } from '../utils/hooks/useMergedRefs';
+
+export type BreadcrumbsProps = {
+  /**
+   * Index of the currently active breadcrumb.
+   * Defaults to last index in the items array.
+   */
+  currentIndex?: number;
+  /**
+   * Breadcrumb items.
+   */
+  items: JSX.Element[];
+  /**
+   * Specify a custom separator element to show between breadcrumb items.
+   * Defaults to the `SvgChevronRight` icon.
+   */
+  separator?: React.ReactNode;
+} & Omit<CommonProps, 'title'>;
+
+/**
+ * A breadcrumb trail is used as a navigational aid to help users keep track
+ * of their place in the application. It is often placed before a page's main content.
+ *
+ * For accessibility, make sure to provide an aria-label describing the type of navigation.
+ *
+ * @example
+ * <Breadcrumbs
+ *   items={[
+ *     <BreadcrumbItem onClick={() => {}}>Root</BreadcrumbItem>,
+ *     <BreadcrumbItem onClick={() => {}}>Item 1</BreadcrumbItem>,
+ *     <BreadcrumbItem onClick={() => {}}>Item 2</BreadcrumbItem>,
+ *   ]}
+ *   aria-label='Primary'
+ * />
+ />
+ */
+export const Breadcrumbs = React.forwardRef(
+  (props: BreadcrumbsProps, ref: React.RefObject<HTMLElement>) => {
+    const {
+      items,
+      currentIndex = items.length - 1,
+      separator,
+      className,
+      ...rest
+    } = props;
+
+    useTheme();
+
+    const breadcrumbsRef = React.useRef<HTMLElement>(null);
+
+    const [hiddenItems, setHiddenItems] = React.useState<JSX.Element[]>([]);
+    const [postItems, setPostItems] = React.useState(() => items.slice(1));
+    const overflowBreakpoints = React.useRef<number[]>([]);
+
+    const [breadcrumbsWidth, setBreadcrumbsWith] = React.useState<number>(0);
+    const updateBreadcrumbsWidth = React.useCallback(
+      ({ width }) => setBreadcrumbsWith(width),
+      [],
+    );
+    const [resizeRef] = useResizeObserver(updateBreadcrumbsWidth);
+    const refs = useMergedRefs(breadcrumbsRef, resizeRef, ref);
+
+    React.useLayoutEffect(() => {
+      if (!breadcrumbsRef.current) {
+        return;
+      }
+
+      // hide items when there's no space available
+      if (
+        breadcrumbsRef.current.offsetWidth < breadcrumbsRef.current.scrollWidth
+      ) {
+        if (postItems.length > 0) {
+          setHiddenItems([...hiddenItems, postItems.shift()!]); // eslint-disable-line @typescript-eslint/no-non-null-assertion
+          setPostItems(postItems);
+          overflowBreakpoints.current.push(breadcrumbsWidth);
+        }
+      }
+      // restore items when there's enough space again
+      else if (
+        overflowBreakpoints.current.length > 0 &&
```

Comment:
Huh? Why you need this variable?

Reply:
It's to keep track of the breakpoints where we hide one or more items, and use those to make them visible again (when resized and there is enough space again).

Reply:
Is it possible that user navigated deeply into one folder, breakpoints were saved, then navigated to completely different folder with different names and old breakpoints will be used with wrong values?

Reply:
Yeah, that's possible. Maybe we need to clear breakpoints when items change?

Reply:
Added another effect hook to reset overflow when items change.

It's located above this main overflow effect hook, so this will re-run and calculate correct widths/breakpoints after items change.

Reply:
I still don't like this variable but it works 😃 Is it possible to do the same but saving just a previous width? If it takes too much time then don't bother, leave as it is because it is working and doesn't affect API.

Reply:
> Is it possible to do the same but saving just a previous width? 

No, I need to save all the widths to be able to restore all of them. The alternative would be recalculating and relayout-ing everything on every single resize.